app.go

```diff
@@ -220,8 +258,14 @@ func (app *App) provide(constructor interface{}) {
 		return
 	}
 	app.logger.PrintProvide(constructor)
+
+	if _, ok := constructor.(Option); ok {
+		app.optionErr = fmt.Errorf("cannot Provide fx.Option: %v", constructor)
+		return
+	}
+
 	if err := app.container.Provide(constructor); err != nil {
-		app.optionErr = multierr.Append(app.optionErr, err)
+		app.optionErr = err
```

Comment:
note that the multierr.Append we have here will never actually "append" because this function *always* short circuits if `optionErr` was non-nil on entry.